testing/integration/agent_long_running_leak_test.go

```diff
@@ -0,0 +1,296 @@
+
+	handles := []processWatcher{}
+
+	// track running beats
+	// the `last 30s` metrics tend to report gauges, which we can't use for calculating a derivative.
+	// so separately fetch the PIDs
+	status, err = runner.agentFixture.ExecStatus(ctx)
+	require.NoError(runner.T(), err)
+	for _, comp := range status.Components {
+		pidStr := regex.FindString(comp.Message)
+		pid, err := strconv.ParseInt(pidStr, 10, 64)
+		require.NoError(runner.T(), err)
+
+		handle, err := sysinfo.Process(int(pid))
+		require.NoError(runner.T(), err)
+		handlesReg := tools.NewSlope(fmt.Sprintf("%s handle usage", comp.Name))
+
+		runner.T().Logf("created handle watcher for %s (%d)", comp.Name, pid)
+		handles = append(handles, processWatcher{handle: handle, pid: int(pid), name: comp.Name, regHandles: handlesReg})
+	}
+
+	testDuration, err := time.ParseDuration(testRuntime)
+	require.NoError(runner.T(), err)
+
+	timer := time.NewTimer(testDuration)
+	defer timer.Stop()
+
+	// time to perform a health check
+	ticker := time.NewTicker(time.Second * 10)
+	defer ticker.Stop()
+
+	done := false
+	start := time.Now()
+	for !done {
+		select {
+		case <-timer.C:
+			done = true
+		case <-ticker.C:
+			err := runner.agentFixture.IsHealthy(ctx)
+			require.NoError(runner.T(), err)
+			// for each running process, collect memory and handles
+			for _, handle := range handles {
+
+				ohc, ok := handle.handle.(types.OpenHandleCounter)
+				if ok {
+					handleCount, err := ohc.OpenHandleCount()
+					require.NoError(runner.T(), err)
+					handle.regHandles.AddDatapoint(float64(handleCount), time.Since(start))
+				}
+
+			}
+		}
+	}
+
+	// we're measuring the handle usage as y=mx+b
+	// if the slope is increasing above a certain rate, fail the test
+	// A number of factors can change the slope during a test; shortened runtime (lots of handles allocated in the first few seconds, producing an upward slope),
+	// filebeat trying to open a large number of log files, etc
+	handleSlopeFailure := 0.1
+
+	for _, handle := range handles {
+		err = handle.regHandles.Run()
+		require.NoError(runner.T(), err)
+
+		runner.T().Logf("=============================== %s (%d)", handle.name, handle.pid)
+		runner.T().Logf("handle formula: %s", handle.regHandles.Formula())
+		handleSlope := handle.regHandles.GetSlope()
+		require.LessOrEqual(runner.T(), handleSlope, handleSlopeFailure, "increase in open handles exceeded threshold: %s", handle.regHandles)
+		runner.T().Logf("===============================")
+	}
+
+	// post-test: make sure that we actually ingested logs.
+	docs, err := estools.GetResultsForAgentAndDatastream(ctx, runner.info.ESClient, "apache.error", status.Info.ID)
+	require.NoError(runner.T(), err, "error fetching apache logs")
+	require.Greater(runner.T(), docs.Hits.Total.Value, 0, "could not find any matching apache logs for agent ID %s", status.Info.ID)
+	runner.T().Logf("Generated %d apache logs", docs.Hits.Total.Value)
```

Comment:
We can avoid stopping at the first assertion failure like:
```suggestion
	docs, err := estools.GetResultsForAgentAndDatastream(ctx, runner.info.ESClient, "apache.error", status.Info.ID)
	if assert.NoError(runner.T(), err, "error fetching apache logs") {
		assert.Greater(runner.T(), docs.Hits.Total.Value, 0, "could not find any matching apache logs for agent ID %s", status.Info.ID)
	}
	
	runner.T().Logf("Generated %d apache logs", docs.Hits.Total.Value)
```

same for the other assertions below